src/openforms/forms/models/form_variable.py

```diff
@@ -0,0 +1,105 @@
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+from ..constants import (
+    FormVariablesDataTypes,
+    FormVariablesInitialValues,
+    FormVariablesSources,
+)
+from .form import Form
+from .form_definition import FormDefinition
+
+
+def get_today():
+    return timezone.now().isoformat()
+
+
+STATIC_INITIAL_VALUES = {FormVariablesInitialValues.today: get_today}
+
+
+class FormVariable(models.Model):
+    form = models.ForeignKey(
+        to=Form,
+        verbose_name=_("form"),
+        help_text=_("Form to which this variable is related"),
+        on_delete=models.CASCADE,
+    )
+    form_definition = models.ForeignKey(
+        to=FormDefinition,
+        verbose_name=_("form definition"),
+        help_text=_(
+            "Form definition to which this variable is related. This is kept as metadata"
+        ),
+        on_delete=models.CASCADE,
+        null=True,
+    )
+    name = models.CharField(
+        verbose_name=_("name"),
+        help_text=_("Name of the variable"),
+        max_length=100,
+    )
+    key = models.SlugField(
+        verbose_name=_("key"),
+        help_text=_("Key of the variable, should be unique with the form."),
+        max_length=100,
+    )
+    source = models.CharField(
+        verbose_name=_("source"),
+        help_text=_(
+            "Where will the data that will be associate with this variable come from"
+        ),
+        choices=FormVariablesSources.choices,
+        max_length=50,
+    )
+    prefill_plugin = models.CharField(
+        verbose_name=_("prefill plugin"),
+        help_text=_("Which, if any, prefill plugin should be used"),
+        blank=True,
+        max_length=50,
+    )
+    prefill_attribute = models.CharField(
+        verbose_name=_("prefill attribute"),
+        help_text=_(
+            "Which attribute from the prefill response should be used to fill this variable"
+        ),
+        blank=True,
+        max_length=50,
+    )
+    data_type = models.CharField(
+        verbose_name=_("data type"),
+        help_text=_("The type of the value that will be associated with this variable"),
+        choices=FormVariablesDataTypes.choices,
```

Comment:
I am kind of undecided if this should have the type of the formio component and use something like https://github.com/open-formulieren/open-forms/blob/master/src/openforms/registrations/contrib/camunda/type_mapping.py#L44 to convert values :thinking:

Reply:
the camunda type mapping approach was a bandaid - ideally, that belongs in `openforms.formio` as part of the #1068.

So, if that sounds favourable, let's make work of it and sort out the Camunda registration backend as well to make use of it.

Reply:
OTOH - for user-defined variables, we cannot infer the data type and will need that information from the form designers. This is also tackled to some degree in Camunda registration backend: https://github.com/open-formulieren/open-forms/blob/master/src/openforms/registrations/contrib/camunda/complex_variables.py

Reply:
Mhmm yes indeed, and then if we replace Formio it would be better if the variables don't depend on the component types used in Formio :thinking: So maybe the frontend should figure out the type of the component and set the `data_type` accordingly